pysat/_instrument.py

```diff
@@ -378,17 +378,17 @@ def __getitem__(self, key):
                     try:
                         # Assume key[0] is integer (including list or slice)
                         return self.data.loc[self.data.index[key[0]], key[1]]
-                    except KeyError:
-                        try:
-                            # Try to force as integer (eg, if ndarray)
-                            idx = self.data.index[key[0].astype(int)]
-                            return self.data.loc[idx, key[1]]
-                        except ValueError:
-                            estring = '\n'.join(("Unable to sort out data.",
-                                                 "Instrument has data : " +
-                                                 str(not self.empty),
-                                                 "Requested key : ", str(key)))
-                            raise ValueError(estring)
+                    # except KeyError:
+                    #     try:
+                    #         # Try to force as integer (eg, if ndarray)
+                    #         idx = self.data.index[key[0].astype(int)]
+                    #         return self.data.loc[idx, key[1]]
+                    except ValueError:
+                        estring = '\n'.join(("Unable to sort out data.",
+                                             "Instrument has data : " +
+                                             str(not self.empty),
+                                             "Requested key : ", str(key)))
+                        raise ValueError(estring)
```

Comment:
Why drop the KeyError detection?  Is it no longer needed?  Why not do:

```
except KeyError or ValueError as err:
```

Reply:
Back when I put #125 together, this was part of the added checks since `pandas.ix` handled all of this internally, and the data access within `_instrument.py` needed to be handled in a more stringent fashion.  The intent was that the first round of checks (KeyError, TypeError) handled most cases where integers were fed into the system, while the second round (removed here) handled a numpy.array of integers.  This was overdesign on my part, as we test for this case and the first round of checks handle it.

I'm fine with leaving it in, but we should test for it and I can't think of a case where it would be needed.  Hence leaving it in as a comment in case we find a bug.

Reply:
Since we have version control and can look this bit up later if needed, I am very much opposed to leaving it in as commented code.  Given what you said I am comfortable having it removed entirely or included in a combined exception statement, as you prefer.

Reply:
ok.  removing completely.

Reply:
Yeah, if we don't have a test for it then it needs to be removed or get a test. I'm a bit late as Jeff already removed it...